component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/retriever/MoesifKeyRetriever.java

```diff
@@ -0,0 +1,305 @@
+            log.error("Failed calling Moesif microservice. Attempted to call url: {}",
+                    MoesifMicroserviceConstants.LIST_URL.replaceAll("[\r\n]", ""),
+                    ex.getMessage().replaceAll("[\r\n]", ""));
+            return;
+        }
+        String authHeaderValue = getAuthHeader(gaAuthUsername, gaAuthPwd);
+        HttpsURLConnection con = null;
+        try {
+            con = (HttpsURLConnection) obj.openConnection();
+            con.setRequestMethod("GET");
+            con.setRequestProperty("Authorization", authHeaderValue);
+            con.setRequestProperty("Content-Type", MoesifMicroserviceConstants.CONTENT_TYPE);
+            con.setReadTimeout(MoesifMicroserviceConstants.REQUEST_READ_TIMEOUT);
+            int responseCode = con.getResponseCode();
+            if (responseCode == HttpsURLConnection.HTTP_OK) {
+                try (BufferedReader in = new BufferedReader(
+                        new InputStreamReader(con.getInputStream(), StandardCharsets.UTF_8));) {
+                    String inputLine;
+                    StringBuffer response = new StringBuffer();
+
+                    while ((inputLine = in.readLine()) != null) {
+                        response.append(inputLine);
+                    }
+                    updateMap(response.toString());
+                }
+            } else if (responseCode >= 400 && responseCode < 500) {
+                log.error("Getting {} from the microservice.", responseCode);
+            } else {
+                throw new APICallException("Getting " + responseCode + " from the microservice and retrying.");
+            }
+        } finally {
+            if (con != null) {
+                con.disconnect();
+            }
+        }
+    }
+
+    /**
+     * Retrieves moesif key of the given orgID by calling the microservice.
+     *
+     * @param orgID
+     * @return moesif key of the orgID
+     * @throws IOException
+     * @throws APICallException
+     */
+    private String callDetailResource(String orgID) throws IOException, APICallException {
+        StringBuffer response = new StringBuffer();
+        String url = null;
+        // Protecting endpoint from SSRF.
+        if (!orgID.isEmpty()) {
+            url = MoesifMicroserviceConstants.DETAIL_URL + "?" + MoesifMicroserviceConstants.QUERY_PARAM + "=" +
+                    orgID;
+        } else {
+            log.error("Failed calling Moesif microservice. Organization ID cannot be empty");
+            return null;
+        }
+        final URL obj;
+        try {
+            obj = new URL(url);
+        } catch (MalformedURLException ex) {
+            log.error("Failed calling Moesif microservice. Attempted to call url: {}", url.replaceAll("[\r\n]", ""),
+                    ex.getMessage().replaceAll("[\r\n]", ""));
+            return null;
+        }
+        String authHeaderValue = getAuthHeader(gaAuthUsername, gaAuthPwd);
+        HttpsURLConnection con = null;
```

Comment:
Even this https connection logic also common right

Reply:
Yes Viraj
Shall I move it ?